### PR TITLE
[risk=low][RW-10991] Extract a CustomizePanel + footer from Runtime Config Panel

### PR DESCRIPTION
**Pull request by jmthibault79**
Also add new RTL tests for these components.

Tested locally by interacting with the runtime config panel through various paths 

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 1]
```diff
@@ -228,31 +189,13 @@ const PanelMain = fp.flow(
       }
     }, [analysisConfig.computeType]);
 
-    const runtimeExists = (status && isVisible(status)) || !!pendingRuntime;
-    const disableControls = runtimeExists && !isActionable(status);
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:232]: moved inside customize-panel

Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.tsx:232]: also true for dataprocExists, unattachedPdExists, and unattachedDiskNeedsRecreate

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 2]
```diff
@@ -228,31 +189,13 @@ const PanelMain = fp.flow(
...
       existingAnalysisConfig.diskConfig.detachable;
-    const unattachedPdExists = !!gcePersistentDisk && !attachedPdExists;
-    const unattachedDiskNeedsRecreate =
-      unattachedPdExists &&
-      analysisConfig.diskConfig.detachable &&
-      (gcePersistentDisk.size > analysisConfig.diskConfig.size ||
-        gcePersistentDisk.diskType !==
-          analysisConfig.diskConfig.detachableType);
-
-    const disableDetachableReason = cond<string>(
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:249]: moved inside disk-selector

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 3]
```diff
@@ -323,64 +266,53 @@ const PanelMain = fp.flow(
 
     const currentRunningCost = machineRunningCost(analysisConfig);
 
-    const diskValidator = {
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:326]: Moved these inside getErrorMessageContent() because they are only used there: diskValidator (inlined), dataprocValidators (inlined), diskErrors, dataprocErrors

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 4]
```diff
@@ -323,64 +266,53 @@ const PanelMain = fp.flow(
...
+          diskSize: diskSizeValidatorWithMessage('standard'),
+        }
+      );
+      const diskErrorDivs = diskErrors ? summarizeErrors(diskErrors) : [];
+
+      const dataprocErrors =
+        analysisConfig.computeType === ComputeType.Dataproc &&
+        validate(
+          { masterDiskSize, workerDiskSize, numberOfWorkers },
+          // We don't clear dataproc config when we change compute type so we can't combine this with the
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:293]: This comment moved from its previous location.  I don't fully understand it.

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 5]
```diff
@@ -392,73 +324,13 @@ const PanelMain = fp.flow(
         analysisConfig.diskConfig.detachable) ||
         (analysisConfig.computeType === ComputeType.Dataproc &&
           !analysisConfig.diskConfig.detachable));
-    // Casting to RuntimeStatus here because it can't easily be done at the destructuring level
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:395]: I guess we must have fixed this at some point, because it's already the right type before casting

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 6]
```diff
@@ -392,73 +324,13 @@ const PanelMain = fp.flow(
...
-    // where we get 'status' from
     const runtimeCanBeUpdated =
       environmentChanged &&
       (
         [RuntimeStatus.RUNNING, RuntimeStatus.STOPPED] as Array<RuntimeStatus>
-      ).includes(status as RuntimeStatus) &&
+      ).includes(runtimeStatus) &&
       runtimeCanBeCreated;
 
-    const renderCreateOrUpdateButton = (
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:404]: I know I just wrote this function, but I changed my mind :)

These are all simple `<Button>`s now

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 7]
```diff
@@ -392,73 +324,13 @@ const PanelMain = fp.flow(
...
-            } else {
-              setPanelContent(PanelContent.ConfirmUpdate);
-            }
-          }}
-          disabled={!runtimeCanBeUpdated}
-        />
-      );
-    };
-
-    const usingDataproc = analysisConfig.computeType === ComputeType.Dataproc;
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:460]: trivial check, not worth the indirection

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 8]
```diff
@@ -584,271 +472,40 @@ const PanelMain = fp.flow(
           [
             PanelContent.Customize,
             () => (
-              <div style={{ marginBottom: '10px' }}>
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:587]: This block goes all the way to L851.  I moved it all to CustomizePanel.

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 9]
```diff
@@ -321,66 +264,49 @@ const PanelMain = fp.flow(
       };
     };
 
-    const currentRunningCost = machineRunningCost(analysisConfig);
-
-    const diskValidator = {
-      diskSize: diskSizeValidatorWithMessage('standard'),
-    };
-
-    const runningCostValidator = {
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:330]: inlined at L268

File: ui/src/app/components/runtime-configuration-panel.tsx (hunk 8 above)
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.tsx:587]: Later: I moved the bottom third to CustomizePanelFooter

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 10]
```diff
@@ -584,271 +463,41 @@ const PanelMain = fp.flow(
...
-                      onClick={() =>
-                        setPanelContent(PanelContent.DeleteRuntime)
-                      }
-                    >
-                      Delete Environment
-                    </LinkButton>
-                    {cond(
-                      [runtimeExists, () => renderNextUpdateButton()],
-                      [
-                        unattachedDiskNeedsRecreate,
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:839]: I removed this condition because it's unreachable.

It requires a logical contradiction: !unattachedPdExists && unattachedDiskNeedsRecreate

File: ui/src/app/components/runtime-configuration-panel/customize-panel.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+      ...defaultAnalysisConfig,
+      computeType: ComputeType.Dataproc,
+    };
+    // TODO why do we need both analysisConfig and existingAnalysisConfig here?
+    await component({ analysisConfig, existingAnalysisConfig });
+    const sparkButton = screen.queryByRole('button', {
+      name: 'Manage and monitor Spark console',
+    });
+    await waitFor(() => expect(sparkButton).toBeInTheDocument());
+    screen.debug();
```
Comment (evrii) [inline ui/src/app/components/runtime-configuration-panel/customize-panel.spec.tsx:303]: Whoops

**Review by evrii — APPROVED** [review]
Looks good thanks for walking through it with me.

File: ui/src/app/components/runtime-configuration-panel.tsx (hunk 1 above)
Comment (NehaBroad) [reply ui/src/app/components/runtime-configuration-panel.tsx:232]: DO you feel we can add comments on some scenarios where isVisible will be true?

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 12]
```diff
@@ -514,29 +389,33 @@ const PanelMain = fp.flow(
                           () => RuntimeStatusRequest.DeleteRuntime,
                         ],
                         [
-                          deletePDSelected && !usingDataproc,
+                          deletePDSelected &&
+                            analysisConfig.computeType !== ComputeType.Dataproc,
                           () => RuntimeStatusRequest.DeleteRuntimeAndPD,
                         ],
                         [
                           // TODO: this configuration is not supported.  Remove?
```
Comment (NehaBroad) [inline ui/src/app/components/runtime-configuration-panel.tsx:397]: Looking at the commen: TODO: this configuration is not supported. Remove? Do we still need this configuration? or will that be part of another PR

**Review by NehaBroad — APPROVED** [review]

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 13]
```diff
@@ -138,7 +99,7 @@ const PanelMain = fp.flow(
       );
     }
 
-    const [status, setRuntimeStatus] = useRuntimeStatus(
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.tsx:141]: ok - did some related renaming

File: ui/src/app/components/runtime-configuration-panel.tsx (hunk 12 above)
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.tsx:397]: I don't want to make any functional changes in this PR.  That's a good candidate for near-future work.

File: ui/src/app/components/apps-panel/expanded-app.tsx [hunk 14]
```diff
@@ -89,13 +89,16 @@ const PauseRuntimeButton = (props: { workspace: Workspace }) => {
     workspace: { namespace, googleProject },
   } = props;
 
-  const [status, setRuntimeStatus] = useRuntimeStatus(namespace, googleProject);
+  const [runtimeStatus, setRuntimeStatusRequest] = useRuntimeStatus(
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/expanded-app.tsx:92]: renaming to reflect types better, similar to elsewhere in this PR

File: ui/src/app/utils/runtime-utils.tsx [hunk 15]
```diff
@@ -975,15 +975,16 @@ export const useRuntimeStatus = (
   RuntimeStatus | undefined,
   (statusRequest: RuntimeStatusRequest) => Promise<void>
 ] => {
-  const [runtimeStatus, setRuntimeStatus] = useState<RuntimeStatusRequest>();
+  const [runtimeStatusRequest, setRuntimeStatusRequest] =
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-utils.tsx:978]: renaming for types
